Sources/NIO/UniversalBootstrapSupport.swift

```diff
@@ -167,6 +178,58 @@ public struct NIOClientTCPBootstrap {
         return NIOClientTCPBootstrap(self.tlsEnablerTypeErased(self.underlyingBootstrap),
                                      tlsEnabler: self.tlsEnablerTypeErased)
     }
+    
+    /// A channel option which can be applied to bootstrap using shorthand notation.
+    /// - See: NIOClientTCPBootstrap.channelOptions(_ options: [Option])
+    public struct Option {
+        private let data: ShorthandOption
+        
+        private init(_ data: ShorthandOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied NIOClientTCPBootstrap
+        /// - Parameter to: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        @usableFromInline
+        func applyOption(to clientBootstrap: NIOClientTCPBootstrap) {
+            data.applyOption(to: clientBootstrap)
+        }
+        
+        fileprivate enum ShorthandOption {
+            case reuseAddr
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to bootstrap: NIOClientTCPBootstrap) {
+                switch self {
+                case .reuseAddr:
+                    _ = bootstrap.channelOption(ChannelOptions.socketOption(.reuseaddr), value: 1)
```

Comment:
this is a bit unfortunate, that we fall back here on the BSD socket options despit the underlying bootstrap also supporting the shorthands...

Reply:
This is addressed in 
https://github.com/apple/swift-nio-transport-services/pull/82
and https://github.com/apple/swift-nio/pull/1489